### PR TITLE
Custom jpsexception

### PR DESCRIPTION
**Pull request by schroedtert**
Instead of using the default std exceptions, a custom exception is added (general purpose, and geometry specific).

### CODE REVIEW

File: cpp/libcore/source/util/jpsexception.cpp [hunk 1]
```diff
@@ -0,0 +1,5 @@
+//
+// Created by tobias on 11.03.21.
```
Comment (zeroset) [inline cpp/libcore/source/util/jpsexception.cpp:2]: Is there a reason for this? :)

Comment (schroedtert) [reply cpp/libcore/source/util/jpsexception.cpp:2]: Not setting up your IDE properly ...

File: cpp/libcore/source/geometry/helper/polygon_helper.cpp [hunk 2]
```diff
@@ -56,21 +57,21 @@ auto jps::geometry::getPolygonCoordinates(std::vector<LineSegment> & p_line_segm
...
            })) {
-        throw std::runtime_error(fmt::format(
+        throw JPSGeometryException(fmt::format(
             FMT_STRING("Could not create Polygon. Not all line segments are on the same level.")));
     }
 
     // sort walls
     try {
         geometry::sortLineSegments(p_line_segments);
     } catch(const std::runtime_error & e) {
```
Comment (zeroset) [inline cpp/libcore/source/geometry/helper/polygon_helper.cpp:67]: Does `geometry::sortLineSegments` still throw `std::runtime_error`?

**Review by zeroset — CHANGES_REQUESTED** [review]

File: cpp/libcore/source/geometry/helper/polygon_helper.cpp (hunk 2 above)
Comment (schroedtert) [reply cpp/libcore/source/geometry/helper/polygon_helper.cpp]: Nope, changed the catch block.

File: cpp/pycore/source/jpscore_binding.cpp [hunk 3]
```diff
@@ -31,4 +32,7 @@ PYBIND11_MODULE(jpscore, m)
     bind_spatial_vector(m_geometry);
     bind_special_area(m_geometry);
     bind_world_builder(m_geometry);
+
+    auto m_util = m.def_submodule("util");
+    bind_jpsexception(m_util);
```
[the remark this answers is not in the document]
Comment (schroedtert) [reply cpp/pycore/source/jpscore_binding.cpp]: Not really, just used the same structure as in libcore part. Moved it to main module.

File: cpp/libcore/source/util/jpsexception.cpp (hunk 1 above)
Comment (schroedtert) [reply cpp/libcore/source/util/jpsexception.cpp]: Removed the file completely, as it was empty anyways

**Review by zeroset — APPROVED** [review]

**Pull request merged**